ml-agents-envs/mlagents/envs/sampler_class.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np 
+from abc import ABC, abstractmethod
+
+from .exception import SamplerException
+
+class Sampler(ABC): 
+
+    @abstractmethod
+    def sample_parameter(self, *args, **kwargs):
+        pass
+
+
+class UniformSampler(Sampler):
+    def __init__(self, param_dict):
+        self.check_keys(param_dict)
+        self.min_value = param_dict["min"]
+        self.max_value = param_dict["max"]
+
+
+    def check_keys(self, param_dict):
+        keys = ["min", "max"]
+        for key in keys:
+            if key not in param_dict:
+                raise SamplerException(
+                    key + " is required to use Uniform Sampling but" 
+                    " this value isn't provided. Please recheck"
+                    " the documentation to ensure that the sampler file"
+                    " matches the required format."
+                )
+
+    def sample_parameter(self):
+        return np.random.uniform(self.min_value, self.max_value)
+    
+
+class MultiRangeUniformSampler(Sampler):
+    def __init__(self, param_dict):
+        self.check_keys(param_dict)
+        self.intervals = param_dict["intervals"]
+    
+    def check_keys(self, param_dict):
+        keys = ["intervals"]
+        for key in keys:
+            if key not in param_dict:
+                raise SamplerException(
+                    key + " is required to use MultiRange Uniform sampler but" 
+                    " this value isn't provided. Please recheck"
+                    " the documentation to ensure that the sampler file"
+                    " matches the required format."
+                )
+
+    
+    def sample_parameter(self):
+        cur_min, cur_max = self.intervals[np.random.randint(len(self.intervals))]
```

Comment:
It's an OK design choice, but I think when you're writing the documentation for this, you should be explicit that each interval is equally likely to be selected (regardless of size). As a user, if I pass [[1,2], [3, 5]], I might expect the latter to be selected twice as often.

Reply:
This wasn't the intended behavior; the new update of code will make the interval selection weighted based on their interval size.